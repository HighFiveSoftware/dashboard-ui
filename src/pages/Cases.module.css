.main_container {
    max-width: 85%;
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    justify-content: space-between;
    /* align-content: space-around; */
}

.charts_container {
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}