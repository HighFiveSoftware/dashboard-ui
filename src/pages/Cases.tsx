import { RouteComponentProps } from '@reach/router';
import React from 'react';
import { Container, Grid } from 'semantic-ui-react';
import { DailyCaseChart } from '../components/DailyCaseChart';
import { DailyCaseStatistics } from '../components/DailyCaseStatistics';
import styles from './Cases.module.css';

const vals = [
  {
    name: 'confirmed',
    value: 1000,
    dif: 10
  },
  {
    name: 'recoveries',
    value: 1000,
    dif: 10
  },
  {
    name: 'deaths',
    value: 1000,
    dif: 10
  }
];

export const Cases = (_props: RouteComponentProps) => {
  return (
    <Container className={styles.main_container}>
      <Grid divided="vertically">
        <Grid.Row>
          <Grid.Column width={8}>
            <DailyCaseStatistics title="Worldwide Cases" vals={vals} />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <DailyCaseChart title="New Cases" />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <DailyCaseChart title="Total Cases" />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Container>
  );
};
