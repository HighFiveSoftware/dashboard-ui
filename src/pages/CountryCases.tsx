import { RouteComponentProps } from '@reach/router';
import { Container, Grid, Message } from 'semantic-ui-react';
import { CaseChart } from '../components/CaseChart';
import { CaseStatistics } from '../components/CaseStatistics';
import styles from './Cases.module.css';
import { capitalize } from '../utils/utils';
import { useCases } from '../hooks/useCases';

interface ICountryCasesProps extends RouteComponentProps {
  country?: string;
}

export const CountryCases = ({ country }: ICountryCasesProps) => {
  const [cases, error, loading] = useCases(country);
  if (loading) return <div>Loading</div>;

  return (
    <Container className={styles.main_container}>
      {error && (
        <Message negative>
          <Message.Header>Error: {error.message}</Message.Header>
        </Message>
      )}

      <Grid divided="vertically">
        <Grid.Row>
          <Grid.Column width={8}>
            <CaseStatistics
              title={`${country ? capitalize(country) : 'Worldwide'} Cases`}
              values={
                cases.length > 0
                  ? [
                      cases[cases.length - 1].confirmedToday,
                      cases[cases.length - 1].deathsToday,
                      cases[cases.length - 1].recoveredToday
                    ]
                  : [0, 0, 0]
              }
            />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <CaseChart title="Total Cases" values={cases} dataKey="today" />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <CaseChart title="New Cases" values={cases} dataKey="change" />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Container>
  );
};
