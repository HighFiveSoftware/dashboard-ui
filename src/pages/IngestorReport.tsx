import React from 'react';
import { Table } from 'semantic-ui-react';
import { RouteComponentProps } from '@reach/router';
import ingestorReports from '../data/ingestorReports.json';

// eslint-disable-next-line no-unused-vars
export const IngestorReport = (props: RouteComponentProps) => (
  <Table celled>
    <Table.Header>
      <Table.Row>
        <Table.HeaderCell>Date</Table.HeaderCell>
        <Table.HeaderCell>Count</Table.HeaderCell>
        <Table.HeaderCell>Time</Table.HeaderCell>
      </Table.Row>
    </Table.Header>

    <Table.Body>
      {ingestorReports.map(
        ({ success, ingestionDate, ingestedCount, ingestionTime }) => (
          <Table.Row positive={success} negative={!success}>
            <Table.Cell>{ingestionDate}</Table.Cell>
            <Table.Cell>{ingestedCount}</Table.Cell>
            <Table.Cell>{ingestionTime} minutes</Table.Cell>
          </Table.Row>
        )
      )}
    </Table.Body>
  </Table>
);
