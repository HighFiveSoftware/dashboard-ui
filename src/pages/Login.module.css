.login_container {
    /* height: 100%; */
    height: 100vh;
    display: flex;
    justify-content: center;
    align-items: center;
}

.login_card {
    max-width: 30%;
}