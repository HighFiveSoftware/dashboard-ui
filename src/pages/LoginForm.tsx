import React, { ChangeEvent, useContext, useState } from 'react';
import {
  Button,
  Form,
  Grid,
  Header,
  InputOnChangeData,
  Message,
  Segment
} from 'semantic-ui-react';
import { RouteComponentProps } from '@reach/router';
import LoginContext from '../context/LoginContext';

interface IFormState {
  email: string;
  password: string;
  error: boolean;
}

const defaultFormState = {
  email: '',
  password: '',
  error: false
};

// eslint-disable-next-line no-unused-vars
export const LoginForm = (props: RouteComponentProps) => {
  const { setLoggedIn } = useContext(LoginContext);

  const [formState, setFormState] = useState<IFormState>(defaultFormState);

  const handleSubmit = () => {
    if (formState.email === 'admin' && formState.password === '123') {
      setFormState(defaultFormState);
      setLoggedIn(true);
    } else {
      setFormState({ ...formState, error: true });
    }
  };

  const handleChange = (
    _e: ChangeEvent<HTMLInputElement>,
    { name, value }: InputOnChangeData
  ) => {
    setFormState({ ...formState, [name]: value });
  };

  return (
    <Grid textAlign="center" style={{ height: '100vh' }} verticalAlign="middle">
      <Grid.Column style={{ maxWidth: 450 }}>
        <Header as="h2" textAlign="center">
          Log-in to your account
        </Header>
        <Form size="large" onSubmit={handleSubmit} error={formState.error}>
          {formState.error && (
            <Message
              error
              header="Login Failed!"
              content="Wrong email or password. Please check again."
            />
          )}
          <Segment stacked>
            <Form.Input
              fluid
              icon="user"
              iconPosition="left"
              placeholder="E-mail address"
              name="email"
              required
              value={formState.email}
              onChange={handleChange}
            />
            <Form.Input
              fluid
              icon="lock"
              iconPosition="left"
              placeholder="Password"
              type="password"
              name="password"
              required
              value={formState.password}
              onChange={handleChange}
            />

            <Button fluid size="large">
              Login
            </Button>
          </Segment>
        </Form>
      </Grid.Column>
    </Grid>
  );
};
