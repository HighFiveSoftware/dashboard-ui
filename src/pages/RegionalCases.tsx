import { RouteComponentProps } from '@reach/router';
import { Grid, Message } from 'semantic-ui-react';
import { CaseChart } from '../components/CaseChart';
import { CaseStatistics } from '../components/CaseStatistics';
import { capitalize } from '../utils/utils';
import { useCases } from '../hooks/useCases';

interface IRegionalCasesProps extends RouteComponentProps {
  region?: string;
}

export const RegionalCases = ({ region }: IRegionalCasesProps) => {
  const [cases, error, loading] = useCases(region);
  if (loading) return <div>Loading</div>;

  return (
    <div>
      {error && (
        <Message negative>
          <Message.Header>Error: {error}</Message.Header>
        </Message>
      )}

      <Grid divided="vertically">
        <Grid.Row>
          <Grid.Column width={8}>
            <CaseStatistics
              title={`${region ? capitalize(region) : 'Worldwide'} Cases`}
              values={
                cases.length > 0
                  ? [
                      cases[cases.length - 1].confirmedToday,
                      cases[cases.length - 1].deathsToday,
                      cases[cases.length - 1].recoveredToday
                    ]
                  : [0, 0, 0]
              }
            />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <CaseChart title="Total Cases" values={cases} dataKey="today" />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row>
          <Grid.Column>
            <CaseChart title="New Cases" values={cases} dataKey="change" />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </div>
  );
};
