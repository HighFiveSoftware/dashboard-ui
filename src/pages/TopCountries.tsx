import { RouteComponentProps } from '@reach/router';
import { Card, Message } from 'semantic-ui-react';
import { TopCountryTable } from '../components/TopCountryTable';
import { useTopCountries } from '../hooks/useTopCountries';

export const TopCountries = (_props: RouteComponentProps) => {
  const [topCountries, error, loading] = useTopCountries();

  if (loading) return <div>Loading</div>;

  return (
    <div>
      {error && (
        <Message negative>
          <Message.Header>Error: {error.message}</Message.Header>
        </Message>
      )}

      <Card fluid>
        <Card.Content>
          <Card.Header>Top Countries</Card.Header>
        </Card.Content>
        <TopCountryTable topCases={topCountries} />
      </Card>
    </div>
  );
};
