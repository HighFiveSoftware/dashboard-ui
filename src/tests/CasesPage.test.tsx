/* eslint-env jest */
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { screen, waitForElementToBeRemoved } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import worldWideData from './data/worldwideData.json';
import countryData from './data/countryData.json';
import { Cases } from '../pages/Cases';
import { renderWithRouterMatch } from './utils';
import { API_URL } from '../constants';

const server = setupServer(
  rest.get(`${API_URL}/cases/`, (req, res, ctx) => {
    const country = req.url.searchParams.get('country');
    if (country) {
      return res(ctx.json(countryData));
    }
    return res(ctx.json(worldWideData));
  })
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

describe('<Cases />', () => {
  test('correctly renders with default route', async () => {
    renderWithRouterMatch(Cases, { path: '/:country?', route: '/' });

    await waitForElementToBeRemoved(() => screen.getByText(/loading/i));

    expect(screen.queryByText(/total cases/i)).not.toBeEmptyDOMElement();
    expect(screen.queryByText(/new cases/i)).not.toBeEmptyDOMElement();
  });

  test('correctly renders with a country route', async () => {
    renderWithRouterMatch(Cases, { path: '/:country?', route: '/turkey' });

    await waitForElementToBeRemoved(() => screen.getByText(/loading/i));

    expect(screen.queryByText(/total cases/i)).not.toBeEmptyDOMElement();
    expect(screen.queryByText(/new cases/i)).not.toBeEmptyDOMElement();
  });

  test('worldwide cases page renders as expected', async () => {
    const component = renderWithRouterMatch(Cases, {
      path: '/:country?',
      route: '/'
    });

    await waitForElementToBeRemoved(() => screen.queryByText(/loading/i));

    expect(component.asFragment()).toMatchSnapshot();
  });

  test('country cases page renders as expected', async () => {
    const component = renderWithRouterMatch(Cases, {
      path: '/:country?',
      route: '/'
    });

    await waitForElementToBeRemoved(() => screen.queryAllByText(/loading/i));

    expect(component.asFragment()).toMatchSnapshot();
  });
});
