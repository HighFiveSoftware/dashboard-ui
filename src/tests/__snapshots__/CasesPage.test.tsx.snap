// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cases /> country cases page renders as expected 1`] = `
<DocumentFragment>
  <div
    class="ui container main_container"
  >
    <div
      class="ui vertically divided grid"
    >
      <div
        class="row"
      >
        <div
          class="eight wide column"
        >
          <div
            class="ui fluid card"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                Worldwide Cases
              </div>
            </div>
            <div
              class="content statistics_container"
            >
              <div
                class="ui tiny statistics"
              >
                <div
                  class="ui statistic"
                >
                  <div
                    class="value"
                  >
                    68,225,435
                  </div>
                  <div
                    class="label"
                  >
                    Confirmed
                  </div>
                </div>
                <div
                  class="ui statistic"
                >
                  <div
                    class="value"
                  >
                    1,556,831
                  </div>
                  <div
                    class="label"
                  >
                    Deaths
                  </div>
                </div>
                <div
                  class="ui statistic"
                >
                  <div
                    class="value"
                  >
                    43,956,174
                  </div>
                  <div
                    class="label"
                  >
                    Recovered
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column"
        >
          <div
            class="ui fluid card"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                Total Cases
              </div>
            </div>
            <div
              class="content"
              style="height: 300px;"
            >
              <div
                class="recharts-responsive-container"
                id="undefined"
                style="width: 100%; height: 100%;"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column"
        >
          <div
            class="ui fluid card"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                New Cases
              </div>
            </div>
            <div
              class="content"
              style="height: 300px;"
            >
              <div
                class="recharts-responsive-container"
                id="undefined"
                style="width: 100%; height: 100%;"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Cases /> worldwide cases page renders as expected 1`] = `
<DocumentFragment>
  <div
    class="ui container main_container"
  >
    <div
      class="ui vertically divided grid"
    >
      <div
        class="row"
      >
        <div
          class="eight wide column"
        >
          <div
            class="ui fluid card"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                Worldwide Cases
              </div>
            </div>
            <div
              class="content statistics_container"
            >
              <div
                class="ui tiny statistics"
              >
                <div
                  class="ui statistic"
                >
                  <div
                    class="value"
                  >
                    68,225,435
                  </div>
                  <div
                    class="label"
                  >
                    Confirmed
                  </div>
                </div>
                <div
                  class="ui statistic"
                >
                  <div
                    class="value"
                  >
                    1,556,831
                  </div>
                  <div
                    class="label"
                  >
                    Deaths
                  </div>
                </div>
                <div
                  class="ui statistic"
                >
                  <div
                    class="value"
                  >
                    43,956,174
                  </div>
                  <div
                    class="label"
                  >
                    Recovered
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column"
        >
          <div
            class="ui fluid card"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                Total Cases
              </div>
            </div>
            <div
              class="content"
              style="height: 300px;"
            >
              <div
                class="recharts-responsive-container"
                id="undefined"
                style="width: 100%; height: 100%;"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="column"
        >
          <div
            class="ui fluid card"
          >
            <div
              class="content"
            >
              <div
                class="header"
              >
                New Cases
              </div>
            </div>
            <div
              class="content"
              style="height: 300px;"
            >
              <div
                class="recharts-responsive-container"
                id="undefined"
                style="width: 100%; height: 100%;"
              >
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
